### PR TITLE
Refactoring - Slim down extension file

### PR DESCRIPTION
**Pull request by katelynienaber**
In this PR:
- Relocated functions/variables from _extension.ts_ not related to starting/stopping extension
- Created _src/globals.ts_ for global variables & their initialization
- Created folders _dataset, uss, job,_ & _shared_ to sort node-related files
- Created new _profiles.ts_ function, _checkCurrentProfile_, for validating credentials
- Moved unit tests related to _checkCurrentProfile_ into _profiles.unit.test_
- Sorted various utils functions into new folder structure

Issue [#652](https://github.com/zowe/vscode-extension-for-zowe/issues/652)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const nls = require('vscode-nls-dev');
 const tsProject = ts.createProject('./tsconfig.json', { typescript });
 
 const inlineMap = true;
-const inlineSource = false;
+const inlineSource = true;
```
Comment (Colin-Stone) [inline gulpfile.js:21]: Not being familiar with gulp what is the effect of this switch?.

File: i18n/sample/package.i18n.json [hunk 2]
```diff
@@ -71,7 +71,7 @@
     "rename": "Rename",
     "copy": "Copy",
     "paste": "Paste",
-    "hMigrateDataSet": "Migrate Data Set",
+    "migrateDataSet": "Migrate Data Set",
```
Comment (Colin-Stone) [inline i18n/sample/package.i18n.json:74]: Was there a rationale regarding the changing of field names?. This was representing the (Hierarchical) Migrate DFSMShsm command and although I prefer the abridged version we run the risk of reducing the meaning of the chosen names particulalrly to z/OS developers looking at the implementation.

Comment (katelynienaber) [reply i18n/sample/package.i18n.json:74]: Okay, I wasn't sure why it was there so I removed it. We can put it back.

File: gulpfile.js (hunk 1 above)
Comment (katelynienaber) [reply gulpfile.js:21]: I was debugging, this is a mistake and can be put back as well :)

File: i18n/sample/package.i18n.json (hunk 2 above)
Comment (Colin-Stone) [reply i18n/sample/package.i18n.json]: No problem. As I say I prefer the 'h' less version but thought I worth calling out that sometimes the fields/variables have been so named to impart additional meaning. So this relates to HMigrate and another new function HRecall will be added soon

**Comment by jellypuno** [comment]
@katelynienaber I added the code but I didn't change the tests :D It's failing. hehe.

File: e [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[33mcommit 59ec0db6deaea832787310db0648530e559aaaef[m
```
Comment (stepanzharychevbroadcom) [inline e:1]: Sorry, but what is this file for?

**Comment by stepanzharychevbroadcom** [comment]
@katelynienaber I checked code and looks fine as for me. We just need to resolve merge conflicts. Also I left a comment about non-understandable file.

**Comment by katelynienaber** [comment]
Reopening due to Jenkins issues

**Pull request closed without merging**